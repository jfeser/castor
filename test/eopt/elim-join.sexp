((queries
  "join(f = g, ascalar(1 as f), ascalar(0 as g))")
 (transforms elim-join-nest elim-join-hash elim-join-filter))
