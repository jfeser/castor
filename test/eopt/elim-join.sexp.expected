$2:
  join((f = g), ascalar(1 as f), ascalar(0 as g))
  depjoin(ascalar(1 as f), atuple([atuple([ascalar(f)], cross),
                                   filter((0.f = g), ascalar(0 as g))], cross))
  depjoin(ascalar(1 as f), select([0.f, g], ahashidx(dedup(select([g as key], ascalar(0 as g))), filter((0.key
                                                                    = g), ascalar(0 as g)), 0.f)))
  
